test_conformance/subgroups/test_workgroup_non_uniform_arithmetic.cpp

```diff
@@ -0,0 +1,774 @@
+            device_, context_, queue_, num_elements_,
+            "test_scinand_non_uniform_logical",
+            scinand_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, SCIN_NU<cl_int, 8>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_scinor_non_uniform_logical",
+            scinor_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, SCIN_NU<cl_int, 9>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_scinxor_non_uniform_logical",
+            scinxor_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, SCEX_NU<cl_int, 7>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_scexand_non_uniform_logical",
+            scexand_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, SCEX_NU<cl_int, 8>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_scexor_non_uniform_logical",
+            scexor_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, SCEX_NU<cl_int, 9>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_scexxor_non_uniform_logical",
+            scexxor_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, RED_NU<cl_int, 7>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_redand_non_uniform_logical",
+            redand_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        error |= test<cl_int, RED_NU<cl_int, 8>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_redor_non_uniform_logical", redor_non_uniform_logical_source,
+            0, useCoreSubgroups_, required_extensions_);
+
+        error |= test<cl_int, RED_NU<cl_int, 9>, G, L>::run(
+            device_, context_, queue_, num_elements_,
+            "test_redxor_non_uniform_logical",
+            redxor_non_uniform_logical_source, 0, useCoreSubgroups_,
+            required_extensions_);
+
+        return error;
+    }
+
+private:
+    cl_device_id device_;
+    cl_context context_;
+    cl_command_queue queue_;
+    int num_elements_;
+    bool useCoreSubgroups_;
+    std::vector<std::string> required_extensions_;
+};
+
+int test_work_group_functions_non_uniform_arithmetic(cl_device_id device,
+                                                     cl_context context,
+                                                     cl_command_queue queue,
+                                                     int num_elements)
+{
+    int error;
```

Comment:
C89 style declaration. Move down to first initialisation.

Reply:
Fixed